datafusion/physical-expr/src/udf.rs

```diff
@@ -38,7 +38,7 @@ pub fn create_physical_expr(
         fun.name(),
         fun.fun(),
         input_phy_exprs.to_vec(),
-        fun.return_type(&input_exprs_types)?,
+        fun.return_type_from_exprs(&input_phy_exprs, input_schema)?,
```

Comment:
@universalmind303 's method helped a lot and it's getting closer.
I think maybe the last issue is here how can we convert `input_phy_exprs` of type &[Arc<dyn PhysicalExpr>] to &[Expr]. 🤔